models/components/versions/3_1_0/component.go

```diff
@@ -0,0 +1,154 @@
+}
+
+func (c Component) GetName() string {
+	return c.Name
+}
+
+func (c Component) GetKey() string {
+	return c.Key
+}
+
+func (c *Component) SetKey(key string) {
+	c.Key = key
+}
+
+func (c Component) GetVerifications() *common.VerificationReferences {
+	return &c.Verifications
+}
+
+func (c Component) GetReferences() *common.GeneralReferences {
+	return &c.References
+}
+
+func (c Component) GetAllSatisfies() []base.Satisfies {
+	// Have to do manual conversion from this Component's Satisfies to the interface base.Satisfies.
+	baseSatisfies := make([]base.Satisfies, len(c.Satisfies))
+	for idx, value := range c.Satisfies {
+		baseSatisfies[idx] = value
+	}
+	return baseSatisfies
+}
+
+func (c Component) GetVersion() semver.Version {
+	return c.SchemaVersion
+}
+
+func (c *Component) SetVersion(version semver.Version) {
+	c.SchemaVersion = version
+}
+
+func (c Component) GetResponsibleRole() string {
+	return c.ResponsibleRole
+}
+
+// Satisfies struct contains data demonstrating why a specific component meets
+// a control
+// This struct is a one-to-one mapping of a `satisfies` item in the component.yaml schema
+// https://github.com/opencontrol/schemas#component-yaml
+type Satisfies struct {
+	ControlKey           string               `yaml:"control_key" json:"control_key"`
+	StandardKey          string               `yaml:"standard_key" json:"standard_key"`
+	Narrative            []NarrativeSection   `yaml:"narrative" json:"narrative"`
+	CoveredBy            common.CoveredByList `yaml:"covered_by" json:"covered_by"`
+	Parameters           []Section            `yaml:"parameters" json:"parameters"`
+	ControlOrigin        string               `yaml:"control_origin" json:"control_origin"`
+	ControlOrigins       []string             `yaml:"control_origins" json:"control_origins"`
+	ImplementationStatus string               `yaml:"implementation_status" json:"implementation_status"`
+}
+
+func (s Satisfies) GetControlKey() string {
+	return s.ControlKey
+}
+
+func (s Satisfies) GetStandardKey() string {
+	return s.StandardKey
+}
+
+func (s Satisfies) GetNarratives() []base.Section {
+	// Have to do manual conversion to the interface base.Section from NarrativeSection.
+	baseSection := make([]base.Section, len(s.Narrative))
+	for idx, value := range s.Narrative {
+		baseSection[idx] = value
+	}
+	return baseSection
+}
+
+func (s Satisfies) GetParameters() []base.Section {
+	// Have to do manual conversion to the interface base.Section from Section.
+	baseSection := make([]base.Section, len(s.Parameters))
+	for idx, value := range s.Parameters {
+		baseSection[idx] = value
+	}
+	return baseSection
+}
+
+func (s Satisfies) GetCoveredBy() common.CoveredByList {
+	return s.CoveredBy
+}
+
+func (s Satisfies) GetControlOrigin() string {
+	return s.ControlOrigin
+}
+
+func (s Satisfies) GetControlOrigins() []string {
+	controlOrigins := set.New()
+	for i := range s.ControlOrigins {
+		controlOrigins.Add(s.ControlOrigins[i])
```

Comment:
Need a test for this: https://codecov.io/gh/opencontrol/compliance-masonry/compare/d140f749617e3f5d8427f4df5e37ad32c57ee96d...2ef2863eee20aad0b7f0f7a2b83aff3990790eac?src=pr#6D6F64656C732F636F6D706F6E656E74732F76657273696F6E732F335F315F302F636F6D706F6E656E742E676F-115